### ISSUE TITLE
Malformed chat-hook state is preventing OTA

### ISSUE THREAD

**Issue by Fang-** [urbit/urbit#3608]
Copying over from #3581 for better visibility:

Trying to pull down the latest OTA fails for me with the following error. Apparently ~palfun hosts chat(s?) with no group associated with it.

https://github.com/urbit/urbit/blob/e17e508379fac7b87a331f15d47bdd72aeec6dbf/pkg/arvo/app/chat-hook.hoon#L912-L918

```
%load-failed
/app/chat-hook/hoon:<[917 3].[918 11]>
/app/chat-hook/hoon:<[915 3].[918 11]>
/app/chat-hook/hoon:<[914 3].[918 11]>
/app/chat-hook/hoon:<[107 25].[107 50]>
/app/chat-hook/hoon:<[107 9].[113 79]>
/app/chat-hook/hoon:<[106 9].[113 79]>
/app/chat-hook/hoon:<[105 9].[113 79]>
/app/chat-hook/hoon:<[103 9].[113 79]>
/app/chat-hook/hoon:<[102 9].[113 79]>
/app/chat-hook/hoon:<[101 9].[113 79]>
/app/chat-hook/hoon:<[100 7].[114 18]>
/app/chat-hook/hoon:<[99 7].[114 18]>
/app/chat-hook/hoon:<[91 7].[114 18]>
/app/chat-hook/hoon:<[90 5].[189 7]>
/app/chat-hook/hoon:<[88 5].[189 7]>
/app/chat-hook/hoon:<[87 5].[189 7]>
/app/chat-hook/hoon:<[86 5].[189 7]>
/app/chat-hook/hoon:<[85 5].[189 7]>
/app/chat-hook/hoon:<[84 5].[355 7]>
/app/chat-hook/hoon:<[83 5].[355 7]>
```

So I poked around the state of these things a bunch. This was kinda painful, chat-hook has no scry endpoints.

In any case, interesting findings. Every locally hosted chat in chat-store has a matching group. For chat-hook however, these two do not have matching groups:

```hoon
{ /~palfun-foslup/dm--sipnes-hanhes
  /~palfun-foslup/dm--naltyc-wornes
}
```

But then, we look at what chats are in chat store, and what groups are in groups-store. We see these entries:

```hoon
{ /~palfun-foslup/dm--~sipnes-hanhes
  /~palfun-foslup/dm--~naltyc-wornes
}
```

Note the `~` in front of the names.

So there is (or at one point was) a bug where the `~` was included erroneously? Which is weird, you'd expect these paths to just get passed around as opaque values, instead of (re)constructed in multiple places. Not sure what's up with that, but userspace devs might be able to take a guess.

Kinda don't want to nuke these DMs, and we probably need to solve this for regular users anyway, so.. what's next?

**Comment by matildepark** [comment]
We're going to have to talk about this today — probably have to amend the on-load for the next OTA and then properly archive these when graph-store gets Chat in it.

**Comment by matildepark** [comment]
Fixed in #3631, #3581.
